web/packages/teleport/src/services/userPreferences/userPreferences.ts

```diff
@@ -35,7 +35,10 @@ export async function getUserPreferences() {
 }
 
 export function updateUserPreferences(preferences: UserPreferencesSubset) {
-  return api.put(cfg.api.userPreferencesPath, preferences);
+  return api.put(cfg.api.userPreferencesPath, {
+    preferences,
+    fieldmask: createFieldMask(preferences),
```

Comment:
As we talked. Frontend should not be concerned about our internal simplementation